source/val/validate_misc.cpp

```diff
@@ -72,6 +72,37 @@ spv_result_t ValidateShaderClock(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+spv_result_t ValidateAssumeTrue(ValidationState_t& _, const Instruction* inst) {
```

Comment:
I think for both of these you need to explicitly check the capability requirement. The parser will allow them if the extension is present, but that is not enough.

Reply:
There's a test covering this already: https://github.com/KhronosGroup/SPIRV-Tools/pull/4161/files#diff-66c8caae78e93e2eee64c56fa87ca7b9230c05103187a9a90fa680a0ce0706cfR115

ValidateSpvExpectAssumeKHR,
       AssumeTrueKHR_RequiresExpectAssumeCapability

ValidateSpvExpectAssumeKHR, ExpectKHR_RequiresExpectAssumeCapabilit